pkgs/applications/editors/vim/plugins/overrides.nix

```diff
@@ -1361,7 +1361,7 @@ assertNoAdditions {
     checkInputs = with self; [
       luasnip
       null-ls-nvim
-      nvim-treesitter
+      nvim-treesitter-legacy
```

Comment:
https://github.com/ray-x/go.nvim/pull/592 implies it is tested/works with main now.